packages/sdk/src/Confidence.ts

```diff
@@ -106,7 +135,15 @@ export class Confidence implements EventSender, Trackable {
    * @internal
    */
   resolve(flagNames: string[]): Promise<FlagResolution> {
-    return this.config.flagResolverClient.resolve(this.getContext(), flagNames);
+    // we first resolve a promise so that if multiple context changes happen in the same tick, we still only make one resolve
+    return Promise.resolve().then(() => {
+      const context = this.getContext();
+      if (!this.flagResolution || !Value.equal(context, this.flagResolution.context)) {
+        this.flagResolution?.abort();
```

Comment:
Should put a reasonable reason in the abort call. Now it's just a generic "opeation was aborted" or something similar.